test/s3path.jl

```diff
@@ -156,6 +156,19 @@ function test_s3_properties(ps::PathSet)
         @test fp2.bucket == "mybucket"
         @test fp2.key == "path/to/some/prefix/"
         @test fp2.version === nothing
+
+        fp3 = S3Path(ps.root.bucket, "/another/testdir/")
+        strs = ["what up", "what else up", "what up again"]
+        write(fp3 / "testfile1.txt", strs[1])
+        write(fp3 / "testfile2.txt", strs[2])
+        write(fp3 / "inner" / "testfile3.txt", strs[3])
+        @test AWSS3.diskusage(fp3) == sum(ncodeunits.(strs))
+
+        # we deliberately pick an older file to compare to so we
+        # can be confident timestamps are different
+        @test AWSS3.lastmodified(fp3) > AWSS3.lastmodified(ps.foo)
+
+        rm(S3Path(ps.root.bucket, "/another/"); recursive=true)  # otherwise subsequent tests may fail
```

Comment:
should use try-finally to make sure we hit the `rm`:
```suggestion
        try
            fp3 = S3Path(ps.root.bucket, "/another/testdir/")
            strs = ["what up", "what else up", "what up again"]
            write(fp3 / "testfile1.txt", strs[1])
            write(fp3 / "testfile2.txt", strs[2])
            write(fp3 / "inner" / "testfile3.txt", strs[3])
            @test AWSS3.diskusage(fp3) == sum(ncodeunits.(strs))
    
            # we deliberately pick an older file to compare to so we
            # can be confident timestamps are different
            @test AWSS3.lastmodified(fp3) > AWSS3.lastmodified(ps.foo)
        finally
            rm(S3Path(ps.root.bucket, "/another/"); recursive=true)  # otherwise subsequent tests may fail
        end
```